Added malloc checks

Melendeze13 opened the pull request.

Melendeze13 committed:
Added malloc checks

imdanielsp commented on src/LinkedList/linked_list.c:
```diff
@@ -28,6 +28,10 @@ pLinkedList makeList() {
   // We need to cast it to a pLinkedList.
   pLinkedList linkedList = (pLinkedList) malloc(sizeof(LinkedList));
+  if (linkedList == NULL) {
+    printf("Malloc failed\n");
```
We don't want to print anything here. Let the client decided what to do. What do you think? (This applies elsewhere).

imdanielsp commented on src/LinkedList/linked_list.c:
```diff
@@ -28,6 +28,10 @@ pLinkedList makeList() {
   pLinkedList linkedList = (pLinkedList) malloc(sizeof(LinkedList));
+  if (linkedList == NULL) {
+    printf("Malloc failed\n");
+    return 0;
```
Although NULL evaluates to 0 (in most systems), lets return the NULL macro so that in other systems where non-0 NULL is represented our library would work.

imdanielsp commented on src/LinkedList/linked_list.c:
```diff
@@ -42,8 +46,16 @@ unsigned int appendToList(pLinkedList linkedList, T data) {
   newNode->data = (T*) malloc(sizeof(T));
+  if (newNode->data == NULL) {
+    printf("Malloc failed\n");
+    return 0;
```
Zero here is good because we are returning an unsigned int, where 0 == fail, 1 == success. Above (in makeList) we are returning a pointer, that's why we want to explicitly return NULL.

imdanielsp commented on src/Queue/queue.c:
```diff
@@ -32,6 +32,10 @@ struct Queue {
   pQueue newQueue = malloc(sizeof(Queue));
+  if (newQueue == NULL) {
+    printf("Malloc failed\n");
+    return 0;
```
Same applies here. Refer to makeList's comments.

imdanielsp reviewed (changes requested):
Good job on catching these bugs! I made some comments, let me know what you think, I believe those are quick fix.

Melendeze13 committed:
DStruct-C: Implemented Stack

Created stack data structure. Created popBack for linkedList. Still needs cleanup.
